components/mcp3008rst

```diff
@@ -0,0 +1,70 @@
+MCP3008 I/O Expander
+=====================
+
+.. seo::
+    :description: Instructions for setting up MCP3008 10 Bit Analog to Digital Converter in ESPHome.
+    :image: mcp3008.png
+
+The Microchip Technology Inc. MCP3008
+devices are successive approximation 10-bit Analogto-Digital (A/D) converters with on-board sample and
+hold circuitry
+
+MCP3008
+--------
+
+The MCP3008 component allows you to use MCP3008 8-Channel 10-Bit A/D Converter
+(`datasheet <http://ww1.microchip.com/downloads/en/DeviceDoc/21295d.pdf>`__,
+`Adafruit <https://www.adafruit.com/product/856>`__) in ESPHome.
+It uses the :ref:`SPI Bus <spi>` for communication.
+
+Once configured, you can use any of the 8 pins as
+sensors for your projects. 
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    MCP3008:
+      - id: 'MCP3008_hub'
+        cs_pin: D8
+
+    # Individual outputs
+    sensor:
+      - platform: mcp3008
+        update_interval: 1s
+        mcp3008: 'mcp3008_hub'
+        id: freezer_temp_source
+        name: "Freezer Temp"
+        number: 0
+      - platform: mcp3008
+        update_interval: 1s
+        mcp3008: 'mcp3008_hub'
+        name: "FridgeTmep"
```

Comment:
Check spelling, also, is this a good example? what will the user see here on the UI?